src/ui/counter.ts

```diff
@@ -66,7 +72,44 @@ export class Counter3D {
       VisibilityComponent.createOrReplace(numberEntity)
       this.digits.push(numberEntity)
 
-      syncEntity(numberEntity, [GltfContainer.componentId], this.id * 20000 + i)
+      syncEntity(numberEntity, [Transform.componentId, GltfContainer.componentId], this.id * 20000 + i)
+    }
+  }
+
+  justifyNumber() {
+    const {
+      components: { Transform }
+    } = getSDK()
+    const numberArr = separateDigits(this.currentNumber)
+    let offset = 0
+
+    switch (this.justify) {
+      case 'left': {
+        offset = 0
+        break
+      }
+      case 'right': {
+        if (!this.zeroPadding) {
+          offset = -(numberArr.length - 1) * this.spacing
+        } else {
+          offset = -(this.digits.length - 1) * this.spacing
+        }
+
+        break
+      }
+      case 'center': {
+        if (!this.zeroPadding) {
+          offset = -((numberArr.length - 1) * this.spacing) / 2
+        } else {
+          offset = -((this.digits.length - 1) * this.spacing) / 2
+        }
+
+        break
+      }
+    }
+
+    for (let i = 0; i < this.digits.length; i++) {
+      Transform.getMutable(this.digits[i]).position = Vector3.create(i * this.spacing + offset, 0, 0)
```

Comment:
this is only being triggered for the active player right ?
Just asking because this transform is being syncronized and if multiple users change it at the same time while the counter is on, it could cause a lot of chaos 😂

Reply:
I think it is okay to merge. It will be the minigame creator's responsibility to only call setNumber() when the player is active anyway.